lib/salestation/web/extractors.rb

```diff
@@ -88,6 +88,75 @@ def merge(other)
         end
       end
 
+      # Handles renaming input keys
+      #
+      # When renaming we want to ensure that the new key provided for the rename does not already
+      # exist in the input. When it does and it has a truthy value, then rename will not happen,
+      # unless `override: true` is also applied. When the new key exists without truthy value,
+      # rename will happen.
+      # By default override is set to false: when input already has value set for the new key,
+      # the old key will be discarded instead of overwriting the value.
+      # For renaming, one should extract both new and old key from then input before to get
+      # expected result, as only then the values can be compared in rename.
+      #
+      # @example
+      #  original_input = {
+      #    'x' => 'a',
+      #    'y' => 'b'
+      #  }
+      #
+      #  extractor = BodyParamExtractor[:x, :y]
+      #    .rename(x: 'z')
+      #  input = {
+      #   'z' => 'a',
+      #   'y' => 'b'
+      #  }
+      #
+      #  extractor = BodyParamExtractor[:x, :y]
+      #    .rename(x: 'y')
+      #  input = {
+      #   'y' => 'b'
+      #  }
+      #  extractor = BodyParamExtractor[:x, :y]
+      #    .rename(x: {new_key: 'y', override: true})
+      #  input = {
+      #   'y' => 'a'
+      #  }
+      class InputRenamer
+        def initialize(extractor, rules)
+          @extractor = extractor
+          @rules = map_rules(rules)
+        end
+
+        def call(rack_request)
+          @extractor
+            .call(rack_request)
+            .map(&method(:rename))
+        end
+
+        def rename(input)
+          @rules.each do |old_key, rule|
+            new_key = rule[:new_key]
+            override = rule[:override]
+
+            if input[new_key].nil? || !input[new_key] || override
```

Comment:
Can you give an example where the _truthy_ check is useful and `nil?` check does not suffice?

Reply:
I can't

Reply:
I was thinking about an empty string, hash or array, but then something at least is provided, I think we can do without it